command/monitor/monitor_test.go

```diff
@@ -0,0 +1,88 @@
+package monitor
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMonitor_Start(t *testing.T) {
+	t.Parallel()
+
+	logger := log.NewInterceptLogger(&log.LoggerOptions{
+		Level: log.Error,
+	})
+
+	m := New(512, logger, &log.LoggerOptions{
+		Level: log.Debug,
+	})
+
+	logCh := m.Start()
+	defer m.Stop()
+
+	go func() {
+		logger.Debug("test log")
+		time.Sleep(10 * time.Millisecond)
+	}()
+
+	for {
```

Comment:
The `for` loop seems redundant here (though not its body.)

Reply:
Similar to `command/monitor/monitor.go`, I copied this file wholesale [from nomad](https://github.com/hashicorp/nomad/blob/master/command/agent/monitor/monitor_test.go), and didn't examine its contents closely.  Again, my limited knowledge of go at this moment means I wouldn't be able to spot these flaws even if I had reviewed it closely.

I'll take a look at addressing your comment, though.

Reply:
Removed